pkg/operator/operator.go

```diff
@@ -89,35 +92,6 @@ func (c OpenShiftControllerManagerOperator) sync() error {
 	if err != nil {
 		return err
 	}
-	// manage status
-	originalOperatorConfig := operatorConfig.DeepCopy()
-	reasonString := ""
-	messageString := ""
-	switch operatorConfig.Spec.ManagementState {
-	case operatorapiv1.Removed:
-		fallthrough
-		// we equally do not allow removed/unmanaged
-	case operatorapiv1.Unmanaged:
-		reasonString = fmt.Sprintf("%sUnsupported", string(operatorConfig.Spec.ManagementState))
-		messageString = fmt.Sprintf("the controller manager spec was set to %s state, but that is unsupported, and has no effect on this condition", string(operatorConfig.Spec.ManagementState))
-
-		// as we are ignoring / not supporting unmanaged/removed, we still process any other inputs to the sync/reconciliation
-		// unlike what we used to do
-	case operatorapiv1.Managed:
-		// we want to empty out the reason/message string if transitioning from the other phases so the default setting above is good
-	}
-
-	for _, condition := range operatorConfig.Status.Conditions {
-		// do not change the current status as part of noting that unmanaged/removed are not supported
-		condition.Reason = reasonString
-		condition.Message = messageString
-		v1helpers.SetOperatorCondition(&operatorConfig.Status.Conditions, condition)
-	}
-	if !equality.Semantic.DeepEqual(operatorConfig.Status, originalOperatorConfig.Status) {
-		if _, err := c.operatorConfigClient.OpenShiftControllerManagers().UpdateStatus(context.TODO(), operatorConfig, metav1.UpdateOptions{}); err != nil {
-			return err
-		}
-	}
```

Comment:
The status sync controller takes care of the CVO status reporting for us if the operator moves to Unamanged or Removed.